test/e2e/ingress/secret.go

```diff
@@ -89,61 +159,129 @@ wrw7im4TNSAdwVX4Y1F4svJ2as5SJn5QYGAzXDixNuwzXYrpP9rzA2s=
 		tls, err := s.ListApisixTls()
 		assert.Nil(ginkgo.GinkgoT(), err, "list tls error")
 		assert.Len(ginkgo.GinkgoT(), tls, 1, "tls number not expect")
+		assert.Equal(ginkgo.GinkgoT(), cert, tls[0].Cert, "tls cert not expect")
+		assert.Equal(ginkgo.GinkgoT(), key_compare, tls[0].Key, "tls key not expect")
+
+		// check DP
+		s.NewAPISIXHttpsClient(host).GET("/ip").WithHeader("Host", host).Expect().Status(http.StatusOK).Body().Raw()
+
+		dialer := &net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}
+
+		http.DefaultTransport.(*http.Transport).DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
I think `DialContext` can also be removed if we already have the correct `ServerName`.

Reply:
Yes, dirty codes.